package.json

```diff
@@ -7,6 +7,12 @@
   "types": "./lib/node/index.d.ts",
   "exports": {
     ".": {
+      "browser": {
+        "types": "./lib/browser/index.d.ts",
+        "require": "./lib/browser/index.js",
+        "import": "./lib/browser/index.mjs",
+        "default": "./lib/browser/index.js"
+      },
       "types": "./lib/node/index.d.ts",
```

Comment:
`types` field must _always be the first_.

Reply:
@kettanaito Why does the types field have to be the first here? It seems like it stops browser.types from being resolved if it's first.

Reply:
https://nodejs.org/api/packages.html#community-conditions-definitions

> "types" - can be used by typing systems to resolve the typing file for the given export. This condition should always be included first.

Doesn't explain why. I haven't tested myself but you might be able to fix your situation if you use something like what I posted in the OP:

```typescript
"customConditions": ["browser"]
```

That is specific to webpack, but if you don't use webpack you can read the [documentation](https://www.typescriptlang.org/tsconfig#customConditions) for it to see if there is an equivalent in whatever you might be using.

To be clear, I haven't tested if this library works in my particular case after the change Artem made, I'm simply pointing out that the solution for me had two parts: this change + the custom condition.

Reply:
I see, I'm assuming that it actually means that types needs to come first within a group of exports, but since browser is separate it can come first. It's just that types needs to come first with the `browser` export config.

Yeah I was using the exact config that you posted `"customConditions": ["browser"]`. There's an issue there because it's resolving types first, when it should be resolving browser.types instead.

Reply:
Yeah I think your original fix here https://github.com/mswjs/interceptors/issues/345#issuecomment-1712962587 is the correct fix. And I think it correctly interprets the line in the docs. From what I know about `customConditions` in tsconfig, is that it'll cause the listed custom conditions to be considered for export resolution.
Reference: https://www.typescriptlang.org/tsconfig#customConditions

Because of that we're supposed to resolve `browser` -> `types` here first.
```json
    ".": {
      "browser": {
        "types": "./lib/browser/index.d.ts",
        "require": "./lib/browser/index.js",
        "import": "./lib/browser/index.mjs",
        "default": "./lib/browser/index.js"
      },
      "types": "./lib/node/index.d.ts",
      "require": "./lib/node/index.js",
      "import": "./lib/node/index.mjs",
      "default": "./lib/node/index.js"
    },
```

Instead with this config, it resolves `types` first which is incorrect.

```json
    ".": {
      "types": "./lib/node/index.d.ts",
      "browser": {
        "types": "./lib/browser/index.d.ts",
        "require": "./lib/browser/index.js",
        "import": "./lib/browser/index.mjs",
        "default": "./lib/browser/index.js"
      },
      "require": "./lib/node/index.js",
      "import": "./lib/node/index.mjs",
      "default": "./lib/node/index.js"
    },
```